### PR TITLE
test(domains): increased code coverage for domains

### PR DESCRIPTION
**Pull request by Mahendra-intel**
## PR Checklist

- [ ] Unit Tests have been added for new changes
- [ ] API tests have been updated if applicable
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

### CODE REVIEW

File: src/app/domains/domains.service.spec.ts [hunk 1]
```diff
@@ -38,14 +38,45 @@ describe('DomainsService', () => {
...
       expect(response).toEqual(domainResponse)
-      done()
     })
     expect(httpClientSpy.get.calls.count()).toEqual(1, 'one call')
   })
 
-  it('should return a single domain object detail when a single record is requested', (done: DoneFn) => {
+  it('should return array of domains when get all domains called ', () => {
+    const domainResponse = {
+      data: [{
```
Comment (rsdmike) [inline src/app/domains/domains.service.spec.ts:49]: the domain response can be re-used for each test, avoid copy/pasting for each function

File: src/app/domains/domains.service.spec.ts [hunk 2]
```diff
@@ -38,14 +38,45 @@ describe('DomainsService', () => {
...
+      }],
+      totalCount: 1
+    }
+
+    httpClientSpy.get.and.returnValue(of(domainResponse))
+
+    service.getData().pipe().subscribe(response => {
+      expect(response).toEqual(domainResponse)
+    })
+    expect(httpClientSpy.get.calls.count()).toEqual(1, 'one call')
```
Comment (rsdmike) [inline src/app/domains/domains.service.spec.ts:64]: lets check to see what it was called with -- we want to ensure the query params are correct

File: src/app/domains/domains.service.spec.ts [hunk 3]
```diff
@@ -92,17 +154,52 @@ describe('DomainsService', () => {
...
+  it('should throw error when a delete request is sent', () => {
+    const domainName = 'testDomain'
+    const error = {
+      status: 404,
+      message: 'Not Found'
+    }
+
+    httpClientSpy.delete.and.returnValue(throwError(error))
+
+    service.delete(domainName).subscribe(null, err => {
```
Comment (rsdmike) [inline src/app/domains/domains.service.spec.ts:201]: subscribe is showing as deprecated when using null, provide a function ()=>{} and i think that should fix it

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/app/domains/domains.service.spec.ts (hunk 1 above)
Comment (Mahendra-intel) [reply src/app/domains/domains.service.spec.ts:49]: done

File: src/app/domains/domains.service.spec.ts (hunk 3 above)
Comment (Mahendra-intel) [reply src/app/domains/domains.service.spec.ts:201]: done

File: src/app/domains/domains.service.spec.ts (hunk 2 above)
Comment (Mahendra-intel) [reply src/app/domains/domains.service.spec.ts:64]: done

File: src/app/domains/domains.service.spec.ts [hunk 4]
```diff
@@ -19,33 +19,64 @@ describe('DomainsService', () => {
     TestBed.configureTestingModule({})
     service = new DomainsService(new AuthService(httpClientSpy as any, routerSpy as Router), httpClientSpy as any)
   })
+  const domainResponse = {
+    data: [{
```
Comment (rsdmike) [inline src/app/domains/domains.service.spec.ts:23]: use domainReq variable below [domainReq]

File: src/app/domains/domains.service.spec.ts [hunk 5]
```diff
@@ -19,33 +19,64 @@ describe('DomainsService', () => {
...
-      totalCount: 1
-    }
+  it('should return array of 25 domains when get all domains called ', () => {
+    httpClientSpy.get.and.returnValue(of(domainResponse))
+
+    service.getData({ pageSize: 25, startsFrom: 0, count: 'true' }).pipe().subscribe(response => {
+      expect(response).toEqual(domainResponse)
+    })
+    const params = httpClientSpy.get.calls.allArgs()[0][0].split('?')[1]
+    expect('$top=25&$skip=0&$count=true').toEqual(params)
```
Comment (rsdmike) [inline src/app/domains/domains.service.spec.ts:52]: the expectation is reversed -- the order should be expect(actual).toEqual(expected)

File: src/app/domains/domains.service.spec.ts [hunk 6]
```diff
@@ -19,33 +19,64 @@ describe('DomainsService', () => {
...
+  it('should return array of domains when get all domains called ', () => {
     httpClientSpy.get.and.returnValue(of(domainResponse))
 
-    service.getData().subscribe(response => {
+    service.getData().pipe().subscribe(response => {
       expect(response).toEqual(domainResponse)
-      done()
     })
+    const params = httpClientSpy.get.calls.allArgs()[0][0].split('?')[1]
+    expect('$count=true').toEqual(params)
```
Comment (rsdmike) [inline src/app/domains/domains.service.spec.ts:63]: same as above

File: src/app/domains/domains.service.spec.ts [hunk 7]
```diff
@@ -19,33 +19,64 @@ describe('DomainsService', () => {
...
 
+  it('should return array of domains when get all domains called ', () => {
     httpClientSpy.get.and.returnValue(of(domainResponse))
 
-    service.getData().subscribe(response => {
+    service.getData().pipe().subscribe(response => {
       expect(response).toEqual(domainResponse)
-      done()
     })
+    const params = httpClientSpy.get.calls.allArgs()[0][0].split('?')[1]
```
Comment (rsdmike) [inline src/app/domains/domains.service.spec.ts:62]: why not use toHaveBeenCalledWith() on httpClientSpy?

File: src/app/domains/domains.service.spec.ts [hunk 8]
```diff
@@ -58,51 +89,86 @@ describe('DomainsService', () => {
...
     service.delete(domainName).subscribe(() => {
-      done()
+    })
+
+    expect(httpClientSpy.delete.calls.count()).toEqual(1)
+  })
+
+  it('should throw error when a delete request is sent', () => {
+    const domainName = 'testDomain'
+    const error = {
```
Comment (rsdmike) [inline src/app/domains/domains.service.spec.ts:163]: same error is created 3 times, can be created once up above

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/app/domains/domains.service.spec.ts (hunk 8 above)
Comment (Mahendra-intel) [reply src/app/domains/domains.service.spec.ts]: done

File: src/app/domains/domains.service.spec.ts (hunk 7 above)
Comment (Mahendra-intel) [reply src/app/domains/domains.service.spec.ts]: done

File: src/app/domains/domains.service.spec.ts (hunk 6 above)
Comment (Mahendra-intel) [reply src/app/domains/domains.service.spec.ts]: done

File: src/app/domains/domains.service.spec.ts (hunk 5 above)
Comment (Mahendra-intel) [reply src/app/domains/domains.service.spec.ts]: done

File: src/app/domains/domains.service.spec.ts (hunk 4 above)
Comment (Mahendra-intel) [reply src/app/domains/domains.service.spec.ts]: done

**Review by rsdmike — APPROVED** [review]

**Pull request merged**